From 51e85647b229aca32c7f35a97a21940121de87bf Mon Sep 17 00:00:00 2001
From: Dinh Nguyen <dinguyen@kernel.org>
Date: Tue, 30 Nov 2021 14:10:44 -0600
Subject: [PATCH 19/20] arm64: dts: n5x: add qspi, usb, and ethernet support

[upstream commit f4c35356e0fce4bbe2994185cc014f13630a1a1d]

commit 619c7250f58862e166e582a1619f89ddcb32e931 from
https://github.com/altera-opensource/linux-socfpga/commits/socfpga-5.4.124-lts

Populate the N5X board dts file with support for QSPI, USB, and
ethernet.

Signed-off-by: Dinh Nguyen <dinguyen@kernel.org>

[Abhishek: Discarded changes from upstream patch those are already present.]

Signed-off-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 .../boot/dts/intel/socfpga_n5x_socdk.dts      | 36 +++++++++++++++++++
 1 file changed, 36 insertions(+)

diff --git a/arch/arm64/boot/dts/intel/socfpga_n5x_socdk.dts b/arch/arm64/boot/dts/intel/socfpga_n5x_socdk.dts
index 60f2f8ca825e..2bc16ba98d05 100644
--- a/arch/arm64/boot/dts/intel/socfpga_n5x_socdk.dts
+++ b/arch/arm64/boot/dts/intel/socfpga_n5x_socdk.dts
@@ -81,6 +81,42 @@
         };
 };
 
+&qspi {
+       status = "okay";
+       flash@0 {
+               #address-cells = <1>;
+               #size-cells = <1>;
+               compatible = "micron,mt25qu02g", "jedec,spi-nor";
+               reg = <0>;
+               spi-max-frequency = <100000000>;
+
+               m25p,fast-read;
+               cdns,page-size = <256>;
+               cdns,block-size = <16>;
+               cdns,read-delay = <2>;
+               cdns,tshsl-ns = <50>;
+               cdns,tsd2d-ns = <50>;
+               cdns,tchsh-ns = <4>;
+               cdns,tslch-ns = <4>;
+
+               partitions {
+                       compatible = "fixed-partitions";
+                       #address-cells = <1>;
+                       #size-cells = <1>;
+
+                       qspi_boot: partition@0 {
+                               label = "Boot and fpga data";
+                               reg = <0x0 0x03FE0000>;
+                       };
+
+                       qspi_rootfs: partition@3FE0000 {
+                               label = "Root Filesystem - JFFS2";
+                               reg = <0x03FE0000 0x0C020000>;
+                       };
+               };
+       };
+};
+
 &uart0 {
 	status = "okay";
 };
-- 
2.31.1

